src/components/tables/DataTable.js

```diff
@@ -177,56 +177,59 @@ export default {
     search () {
       this.page = 1
     },
-    page () {
-      this.update()
+    filteredItems: {
+      handler: function () {
+        if (this.everyItem) this.all = true
+      },
+      deep: true
+    },
+    selected () {
+      if (this.everyItem) this.all = true
     }
   },
 
   methods: {
-    update () {
-      this.$emit('update', {
-        page: this.page,
-        rowsPerPage: this.rowsPerPage,
-        sorting: this.sorting,
-        desc: this.desc
-      })
+    isSelected (item) {
+      return this.selected && this.selected.find(i => i[this.selectedKey] === item[this.selectedKey])
     },
     sort (index) {
```

Comment:
I think that this method could be cleaned up a bit. The same call is made for every scenario, the only thing that changes is the object that is emitted. Assign the value of that object and only have one this.$emit.